packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_client.dart

```diff
@@ -16,25 +21,78 @@ import 'package:built_collection/built_collection.dart';
 /// {@endtemplate}
 class EndpointClient {
   /// {@macro amplify_analytics_pinpoint_dart.endpoint_client}
-  EndpointClient(
-    this._appId,
-    this._fixedEndpointId,
-    this._pinpointClient,
-    this._globalFieldsManager,
-    this._endpointBuilder,
-  );
-
-  final String _appId;
-  final PinpointClient _pinpointClient;
+  @visibleForTesting
+  EndpointClient({
+    required String pinpointAppId,
+    required PinpointClient pinpointClient,
+    DeviceContextInfo? deviceContextInfo,
+  })  : _pinpointAppId = pinpointAppId,
+        _pinpointClient = pinpointClient,
+        _endpointBuilder = PublicEndpoint(
+          effectiveDate: DateTime.now().toUtc().toIso8601String(),
+          demographic: EndpointDemographic(
+            appVersion: deviceContextInfo?.appVersion,
+            locale: deviceContextInfo?.locale,
+            make: deviceContextInfo?.make,
+            model: deviceContextInfo?.model,
+            modelVersion: deviceContextInfo?.modelVersion,
+            platform: deviceContextInfo?.platform?.name,
+            platformVersion: deviceContextInfo?.platformVersion,
+            timezone: deviceContextInfo?.timezone,
+          ),
+          location: EndpointLocation(
+            country: deviceContextInfo?.countryCode,
+          ),
+        ).toBuilder();
 
-  final String _fixedEndpointId;
+  /// {@macro amplify_analytics_pinpoint_dart.endpoint_client}
+  @visibleForTesting
```

Comment:
I don't see this used in a test.

Reply:
I will make unit tests for the followup PR to refactor Endpoint Info Management